scrapy/signalmanager.py

```diff
@@ -1,12 +1,39 @@
 from __future__ import absolute_import
-from pydispatch import dispatcher
+import warnings
+
 from scrapy.utils import signal as _signal
+from scrapy.exceptions import ScrapyDeprecationWarning
+from scrapy.dispatch.utils import robust_apply as _robust_apply
+from scrapy.dispatch.utils import func_accepts_kwargs
+from scrapy.dispatch import Signal as _Signal
```

Comment:
(Reposting this here because I accidentally attached it to an old commit ;)) Is there a reason to add the leading underscore?